infra/gravity/testdata/status-degraded-1521.json

```diff
@@ -0,0 +1,129 @@
+{
+  "cluster": {
+    "application": {
+      "repository": "gravitational.io",
+      "name": "telekube",
+      "version": "6.1.27"
+    },
+    "state": "active",
```

Comment:
The [1521](https://github.com/gravitational/gravity/issues/1521) is about a different situation. The cluster state should never be active with failed probes (as in this case below). The issue is about the state when the cluster is healthy (i.e. no failed probes) but the cluster controller (the active `gravity-site` Pod) did not reset the cluster state to `active` since its own state checker loop has yet to determine the cluster state.

Reply:
I will update the test and filename.

> The cluster state should never be active with failed probes (as in this case below). 

Are you aware of a gravity issue tracking the inconsistency seen in this status?  Since this isn't 1521 but it sounds like it is a real problem, I'd like to link back to the correct issue.

Reply:
No, I think there's a bit of misunderstanding. `gravity status` should already work as expected w.r.t cluster state being degraded with failed probes. What [1521](https://github.com/gravitational/gravity/issues/1521) additionally describes is when the cluster state is degraded although the agent (planet) state is healthy. Since `gravity status` pulls the agent's status directly, it evaluates it in the form of node statuses. Hence, it planet reports healthy there will be no failed probes in the `gravity status` output.
The problem is then that `gravity status` does not control the cluster's state field - active `gravity-site` Pod does and it can still keep the cluster `degraded` (even when it is actually healthy - according to agent info) since it has its own loop that uses the same planet agent.
If `gravity status` pulled the info from planet  before `gravity-site` Pod did and the cluster was previously degraded, `gravity status` still reads degraded from cluster state and only when `gravity-site`'s loop pulls the state and sees that it's healthy will it reset the cluster state.
As long as we're waiting for `gravity-site` to reset the state, `gravity status` shows `degraded` for no obvious reason.
Hope that explains it.

Reply:
Given we've already gone back & forth a couple times, I'd like to discuss this real time - perhaps Monday?

> gravity status should already work as expected w.r.t cluster state being degraded with failed probes. 

I pulled the status json in this file from robotest/gravity 6.1.27 the wild:  see [this query](https://console.cloud.google.com/logs/viewer?authuser=0&expandAll=false&project=kubeadm-167321&minLogLevel=0&timestamp=2020-05-29T22:34:05.422000000Z&customFacets=&limitCustomFacetWidth=true&interval=PT1H&scrollTimestamp=2020-05-27T21:12:12.519955220Z&dateRangeEnd=2020-05-29T22:34:03.251Z&advancedFilter=resource.type%3D%22project%22%0Alabels.__uuid__%3D%22ea6b230b-a447-4fc7-8128-6a2f3f1a9dce%22%0Alabels.__suite__%3D%22a9601994-7ead-43f0-ba76-b7cc497c3660%22%0AjsonPayload.cmd%3D%22sudo%20gravity%20status%20--output%3Djson%20--system-log-file%3Dgravity-system.log%22%0AAND%20jsonPayload.message:%22%5C%22state%5C%22:%5C%22active%5C%22,%22%0AAND%20jsonPayload.message:%22%5C%22system_status%5C%22:2,%22&dateRangeUnbound=backwardInTime) for an example.  Strip `uuid` and `suite` from the query and there are further examples.

Is this observed status inconsistency between `state`, `system_status` and `nodes` a bug in gravity?

If the inconsitency is a gravity bug: Does we already have an issue #? I'd like to link it in the code. I can create one if there isn't an issue.

If the status here is not a gravity bug: I don't understand what your original comment https://github.com/gravitational/robotest/pull/233#discussion_r432571825 is suggesting.  We should probably talk about it in a higher bandwidth medium.

Reply:
Renamed to https://github.com/gravitational/gravity/issues/1641 in 936a6c62.